### ISSUE TITLE
Switch RegisterLegacyAminoCodec -> RegisterAminoMsg

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/osmosis#1106]
cref https://github.com/cosmos/cosmos-sdk/issues/10870 

Basically we should be using `RegisterAminoMsg` in order to more quickly catch the problems with large amino routes on ledgers.

Blocked on us updating to a compatible SDK version.

**Comment by mattverse** [comment]
I love this solution! Should we aim to implement it first in our sdk fork or should we be aiming to create a PR directly to cosmos sdk?

**Comment by ValarDragon** [comment]
Its already done in the SDK here! https://github.com/cosmos/cosmos-sdk/pull/11349

We really just need to cherry-pick the merge commit from that branch onto our SDK fork, then update this repo! The SDK did the full update!

**Comment by mattverse** [comment]
Oh nice! This is amazing! Totally necessary! Assigning myself!

**Comment by p0mvn** [comment]
Hey @mattverse @AlpinYukseloglu . Seeing you assigned, just wanted to ask what is the status on this? Could one of you update the labels and status on the project board, please

**Comment by AlpinYukseloglu** [comment]
Thanks for catching this @p0mvn -  looks like this one got lost in the backlog. @ValarDragon do you think this is a priority for our current sprint? If yes I'll take a crack at it asap

**Comment by mattverse** [comment]
@AlpinYukseloglu Since we just need to cherry pick the implementation done in the upstream, I think you can go ahead and shoot it when you have extra time

**Comment by ValarDragon** [comment]
Finally got fixed I think, per @mattverse slack msg
